libstuff/STCPNode.cpp

```diff
@@ -332,11 +359,79 @@ void STCPNode::Peer::sendMessage(const SData& message) {
 }
 
 void STCPNode::Peer::closeSocket(STCPManager* manager) {
-    lock_guard<decltype(socketMutex)> lock(socketMutex);
+    lock_guard<decltype(_stateMutex)> lock(_stateMutex);
     if (s) {
         manager->closeSocket(s);
         s = nullptr;
     } else {
         SWARN("Peer " << name << " has no socket.");
     }
 }
+
+string STCPNode::Peer::responseName(Response response) {
+    switch (response) {
+        case STCPNode::Peer::Response::NONE:
+        return "NONE";
+        break;
+        case STCPNode::Peer::Response::APPROVE:
+        return "APPROVE";
+        break;
+        case STCPNode::Peer::Response::DENY:
+        return "DENY";
+        break;
+    }
+    return "";
+}
+
+void STCPNode::Peer::setCommit(uint64_t count, const string& hashString) {
+    lock_guard<decltype(_stateMutex)> l(_stateMutex);
+    const_cast<atomic<uint64_t>&>(commitCount) = count;
+    hash = hashString;
+}
+
+void STCPNode::Peer::getCommit(uint64_t& count, string& hashString) {
+    lock_guard<decltype(_stateMutex)> l(_stateMutex);
+    count = commitCount.load();
+    hashString = hash.load();
+}
+
+STable STCPNode::Peer::getData() const {
+    // Add all of our standard stuff.
+    STable result({
+        {"name", name},
+        {"host", host},
+        {"state", (stateName(state) + (connected() ? "" : " (DISCONNECTED)"))},
+        {"latency", to_string(latency)},
+        {"nextReconnect", to_string(nextReconnect)},
+        {"id", to_string(id)},
+        {"failedConnections", to_string(failedConnections)},
+        {"loggedIn", (loggedIn ? "true" : "false")},
+        {"priority", to_string(priority)},
+        {"version", version},
+        {"hash", hash},
+        {"commitCount", to_string(commitCount)},
```

Comment:
should this use `commitCount.load()`?

Reply:
Don't need to, it will attempt to cast to an appropriate type, which will call this: https://en.cppreference.com/w/cpp/atomic/atomic/operator_T